### DISCUSSION TITLE
Client schema extension

### DISCUSSION THREAD

**Discussion by Meemaw** [urql-graphql/urql#3566] [Q&A]
Is it possible to do client side schema extension? Similar to https://relay.dev/docs/guided-tour/updating-data/client-only-data/#:~:text=Client%2DOnly%20Data%20(Client%20Schema,and%20updated%20on%20the%20client?

**Comment by kitten** [comment]
No, if you have client-parts to a schema, we currently expect you to create an intermediary schema SDL output, i.e. a `schema.graphql` file that has the changes added to it that you need for that specific client.

In theory it's possible for you to combine these in types as well, but I haven't tested this yet, so this isn't strictly supported.

It's kind of similar actually in Relay, I suppose, where they need you to add these SDL files for the final schema. We could consider loading extensions to a schema in the CLI/LSP dynamically, I suppose 🤔
